### PR TITLE
Fixes spot light frustum optimizatin when inside scene bbox

### PR DESCRIPTION
**Pull request by Kuranes**
- better near/far bounding, use of those for the optimized case
- more robust transformation handling for spot
- DRY code (near/far, lightup, single epsilon)

### CODE REVIEW

File: sources/osgShadow/ShadowMap.js [hunk 1]
```diff
@@ -102,8 +102,8 @@ define( [
 
                 //Matrix.clampProjectionMatrix( m, nv.getComputedNear(), nv.getComputedFar(), nv.getNearFarRatio() );
                 // we should use accessors when touching external object !!!
-                this._shadowTechnique._nearCaster = nv.getComputedNear();
-                this._shadowTechnique._farCaster = nv.getComputedFar();
+                this._shadowTechnique._depthRange[ 0 ] = nv.getComputedNear();
```
Comment (cedricpinson) [inline sources/osgShadow/ShadowMap.js]: use accessors when field are +1 indirection

**Comment by cedricpinson** [comment]
is it possible to test this issue, to lock the code. I am sure what was the problem and how it is solved

**Comment by Kuranes** [comment]
Wasn't something that broke, but one use case not tested in the demo.
Could add pure "data" test from use case into unit tests to prevent regression ?

**Comment by cedricpinson** [comment]
I the data are small yes, but it must be a unittest, not something loading a scene ...

**Pull request merged**